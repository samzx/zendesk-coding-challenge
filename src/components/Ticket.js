import React from 'react';

import Details from './Details';
import Conversation from './Conversation';

import { url, comments, tickets } from '../app';

class Ticket extends React.Component {
    state = {
        comments: null,
        errors: null,
        loading: false
    }

    componentDidMount() {
        this._mounted = true;
        this.setState({ loading: true })
        fetch(url + comments + `/${this.props.currentTicket}`, {
            method: 'GET',
        })
        .then((res) => {
            if(res.ok) {
                return res.json();
            } else {
                throw res;
            }
        })
        .then((json) => {
            if(!this._mounted) return;
            this.setState({ comments: json.comments, loading: false });
        })
        .catch((e) => {
            if(!this._mounted) return;
            this.setState({ errors: `${e.status} ${e.statusText}`, loading: false });
            console.log(e);
        });
    }

    componentWillUnmount() {
        this._mounted = false;
    }

    render() {
        return(
            <div className="ticket">
                <Details
                    currentTicket={this.props.currentTicket}
                    data={this.props.data}
                />
                <Conversation comments={this.state.comments} loading={this.state.loading} />
            </div>
        );
    }
}

export default Ticket;