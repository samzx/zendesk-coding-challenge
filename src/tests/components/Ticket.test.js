import React from "react";
import { shallow } from "enzyme";
import Ticket from "../../components/Ticket";

describe("<Ticket />", () => {
  it("should render Ticket correctly", () => {
    const wrapper = shallow(<Ticket />);
    expect(wrapper).toMatchSnapshot();
  });
});
