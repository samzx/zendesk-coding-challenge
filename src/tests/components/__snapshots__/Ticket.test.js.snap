// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Ticket /> should render Ticket correctly 1`] = `
<div
  className="ticket"
>
  <div
    className="ticket--heading"
  >
    <h2>
      Details
    </h2>
  </div>
  <Details />
</div>
`;
